scripts/check_frontend_test_coverage.py

```diff
@@ -326,6 +325,7 @@
     'svm-prediction.service.ts',
     'teacher.ts',
     'teacher2.ts',
+    'text_classifier.ts',
```

Comment:
We are not expecting to add new filenames in this file. Can you please add a test for this new file? :)

Reply:
is this an autogeneratedFile?

Reply:
Lets please not add to this file...

Reply:
Yes, this is auto generated file. Hence, no unit tests are available for it.

Reply:
We ar expecting this place/list to be the list of files that needs coverage of 100%, can we avoid generating coverage for this folder?

@nithusha21, Do you see any other solution?

@srijanreddy98 We had looked into this, right? DO you remember how we were able to avoid generating coverage for unexpected dirs.?

Reply:
I talked with @prasanna08 about this. We should do the following:

- Have a list constant in this file that holds AUTOGENERATED_DIR_FILEPATHS; files in these directories will not fall under the coverage checks.
- Have a backend test that validates that every element of AUTOGENERATED_DIR_FILEPATHS corresponds to a line in the .gitignore file.

That should be foolproof, I think. @DubeySandeep does that sound good?

Reply:
I'm fine with that but in case we find a way to not generate code-coverage for the .gititnore files dirs then I think it would better i.e, we don't have to manage another set of lists and test for that list.  (Cause currently, we are somehow generating coverage for node_modules dir as well so I think instead of spending time on generating coverage and then ignoring it we should find a way to avoid generating the coverage in the first place.)

Reply:
I think we should have an explicit list since we really shouldn't be using the .gitignore file directly like that -- no one will know it has side effects when editing it. Maybe node_modules can just be included explicitly in the exclusion list too, would that work? (But if it's actually being evaluated then why do we still have full coverage currently?)

Reply:
@seanlip @DubeySandeep @nithusha21 , my method to not include node_modules was far from the optimal approach. My approach was to ignore the coverage generated for node_modules. The thing we want is to not generate a report at all. karma.conf has an option which can do this. Its called `exclude`. We can use to ignore files and folders. We already use this option. https://github.com/oppia/oppia/blob/develop/core/tests/karma.conf.ts#L50-L54

Reply:
> I think we should have an explicit list since we really shouldn't be using the .gitignore file directly like that -- no one will know it has side effects when editing it. Maybe node_modules can just be included explicitly in the exclusion list too, would that work? (But if it's actually being evaluated then why do we still have full coverage currently?)

Karma only tests ts files (I think). Most of the libs ship compiled code and when angular encounters the libs for the first time in the compile/ build stage, it converts them to es modules. That's why there are no files that karma includes in node_modules until recently. I introduced a third-party library (ngx-leaflet) that included a demo folder in the package too. That was being checked for coverage.

Reply:
@prasanna08 Can you please include the autogenerated dir in the [exclude](https://github.com/oppia/oppia/blob/develop/core/tests/karma.conf.ts#L50-L54) section?

Reply:
@prasanna08 -- just wanted to confirm that I'm fine with whatever @srijanreddy98 and @DubeySandeep concur on. Please let us know if you have questions about this. Thanks!

Reply:
So, I tried adding path to exclude section of karma.conf but the files were still counted by the coverage checker. The paths which I tried are `local_compiled_js/extensions/classifiers/proto/*` and `extensions/classifiers/proto/*` but neither of them worked. I asked @srijanreddy98 but yet to hear back from him.

Reply:
Hi @srijanreddy98 @DubeySandeep, I tried adding exception for path `extesionsions/classifiers/proto/*.ts` and `extesionsions/classifiers/proto/*.js` in the exclude section of karma.conf.ts but it didn't work. Even after that, when I run frontend tests with --check_coverage argument, text_classifier.ts is shown to be not completely covered. Can either of you confirm whether what I am doing is correct or how does exclude work?

Reply:
I tried checking the config and found [exclude field won't work](http://karma-runner.github.io/6.3/config/configuration-file.html#exclude).

@srijanreddy98 Can you please suggest whether we are expecting to add it there or somewhere else?

Reply:
I talked to @prasanna08 yesterday. This was supposed to be a simple change but now it needs some debugging. For the moment I have asked Prasanna to use the mechanism I had introduced to exclude node_modules from our python coverage checks. By the end of this month, I'll take a look again at this. Thanks!

Reply:
@srijanreddy98 Thanks for looking into it, I've created an issue at #12733 for tracking this task and assigned it to you!